nav2_navfn_planner/include/nav2_navfn_planner/navfn_planner.hpp

```diff
@@ -111,33 +120,35 @@ class NavfnPlanner : public rclcpp::Node
   // Determine if a new planner object should be made
   bool isPlannerOutOfDate();
 
+  std::unique_ptr<nav2_robot::Robot> robot_;
+
   // Planner based on ROS1 NavFn algorithm
   std::unique_ptr<NavFn> planner_;
 
   // Service client for getting the costmap
-  nav2_tasks::CostmapServiceClient costmap_client_;
+  nav2_util::CostmapServiceClient costmap_client_{"navfn_planner"};
 
   // Publishers for the path and endpoints
-  rclcpp::Publisher<nav_msgs::msg::Path>::SharedPtr plan_publisher_;
-  rclcpp::Publisher<visualization_msgs::msg::Marker>::SharedPtr plan_marker_publisher_;
+  rclcpp_lifecycle::LifecyclePublisher<nav_msgs::msg::Path>::SharedPtr plan_publisher_;
+  rclcpp_lifecycle::LifecyclePublisher<visualization_msgs::msg::Marker>::SharedPtr
+    plan_marker_publisher_;
 
-  // The costmap to use
+  // The costmap to use and its size
   nav2_msgs::msg::Costmap costmap_;
   uint current_costmap_size_[2];
 
   // The global frame of the costmap
-  std::string global_frame_;
+  const std::string global_frame_{"map"};
 
   // Whether or not the planner should be allowed to plan through unknown space
-  bool allow_unknown_;
+  const bool allow_unknown_{true};
 
-  // Amount the planner can relax the space constraint
+  // If the goal is obstructed, the tolerance specifies how many meters the planner
+  // can relax the constraint in x and y before failing
   double tolerance_;
```

Comment:
We should be able to define different tolerances for `x` and `y`

Reply:
@mhpanah Ok. Can you enter an issue for that? Don't want to make that change in this PR.

Reply:
@mjeronimo sure. will do.